docs/controllers/reconciler.md

```diff
@@ -1,29 +1,213 @@
+    W
+    S
+    end
+    subgraph "Application"
+    Controller
+    R
+    end
+```
+
+A [Controller] contains an internal machinery that will:
+
+- watch the relevant api endpoints in Kubernetes
+- map/correlate changes from those apis into your [[object]]
+- schedule reconciliations (no parallel reconciliations of same object)
+- call out to your `reconcile` function
+- take the result of that reconciliation to decide when to reschedule
+
+Notably, it's more than a simple watch that's glued to your reconciler. It's an infinite loop with tolerance for a wide class of failures, and a user-specifiable mechanism to correlate changes from different objects to your object.
+
+## The World
+
+The state of the world is generally one or more Kubernetes objects, possibly linked with [ownerReferences], but this is not always the case.
+
+We say **state of the world** because you could control things **outside Kubernetes**; you could be dependent on some external api that you have semantically linked to your cluster. You could have populated this API from a custom resource - in which case you may want [finalizers] to ensure the api gets cleaned up on CRD deletion - or you could simply use this external API as an auxillary source of information.
+
+The **relations** between your main [[object]] and **child objects** needs to be clearly defined for a controller to be most effective. These relations are defined with either [Controller::owns] or [Controller::watches]:
+
+### Owned Relation
+
+The [Controller::owns] relation is the most straight-forward and most ubiquitous one. One object controls the lifecycle of a child object, and cleanup happens automatically via [ownerReferences].
+
+```rust
+let cmgs = Api::<ConfigMapGenerator>::all(client.clone());
+let cms = Api::<ConfigMap>::all(client.clone());
+
+Controller::new(cmgs, ListParams::default())
+    .owns(cms, ListParams::default())
+```
+
+This [configmapgen example](https://github.com/kube-rs/kube-rs/blob/master/examples/configmapgen_controller.rs) uses one custom resource `ConfigMapGenerator` whose controller is in charge of the lifecycle of the child `ConfigMap`.
+Delete the `ConfigMapGenerator` instance? Kubernetes will automatically cleanup the associated `ConfigMap`, and the controller association/correlation is done using Kubernetes' own [ownerReferences] with no manual mapping logic required.
+
+### Watched Relations
+
+The [Controller::watches] relation is for related Kubernetes objects **without** [ownerReferences], i.e. without a standard way for the controller to map the object to the root object. Thus, you need to define this mapper yourself:
+
+```rust
+let main = Api::<MainObj>::all(client);
+let related = Api::<RelatedObject>::all(client);
+
+let mapper = |obj: RelatedObject| {
+    obj.spec.object_ref.map(|oref| {
+        ReconcileRequest::from(oref)
+    })
+};
+
+Controller::new(main, ListParams::default())
+    .watches(related, ListParams::default(), mapper)
+```
```

Comment:
Related to the question in https://discord.com/channels/500028886025895936/951051836972994610

Think this SHOULD work.